modules/kubernetes-addons/cert-manager/data.tf

```diff
@@ -0,0 +1,32 @@
+data "aws_route53_zone" "selected" {
+  for_each = contains(var.domain_names, "*") ? {} : { for domain in toset(var.domain_names) : domain => domain }
+
+  name = each.key
+}
+
+
+data "aws_iam_policy_document" "cert_manager_iam_policy_document" {
+  statement {
+    effect    = "Allow"
+    resources = ["*"]
```

Comment:
https://cert-manager.io/docs/configuration/acme/dns01/route53/#set-up-an-iam-role
```suggestion
    resources = ["arn:aws:route53:::change/*"]
```

Reply:
It is for ListHostedZonesByName, not GetChange, so it might be probably yet another resource. The documentation is explicit about "*" and I found in API documentation that there is no dedicated resource for this action so most likely it have to be "*".

Reply:
This seems to be valid per the IAM console - what error are you seeing, is it a cert-manager error or invalid IAM policy error?

![image](https://user-images.githubusercontent.com/10913471/168296083-29352ccf-6133-431e-a608-a8253b7bcf2a.png)

Reply:
Ok, I think there is only way to check it: just a do full E2E test with `arn:aws:route53:::change/*` resource. I just verified it works with `*` and I've got correct Certificate. If I'll get the same result with this modification then I'll commit it.

Reply:
The Certificate is generated correctly and I see GetChange events in Event history without any errors so I guess it is correct IAM policy.